src/diagnostics/lintstore.md

```diff
@@ -0,0 +1,40 @@
+# LintStore
+
+This page documents some of the machinery around lint registration and how we
+run lints in the compiler.
+
+The `LintStore` is the central piece of infrastructure, around which everything
+rotates. It's not available during the early parts of compilation (i.e., before
+TyCtxt) in most of the code, as we need to fill it in with all of the lints,
+which can only happen after plugin registration.
```

Comment:
Can you describe where the `LintStore` is created and how it eventually gets frozen? IIRC, it is created by the plugin registrar (where it is mutable), but then it is "frozen" (put into a `Rc`) and given to the tcx?